### ISSUE TITLE
Hangouts v17 forced update notice

### ISSUE THREAD

**Issue by AusHick** [kmark/XHangouts#24]
Is it possible to block the full-screen update notice that has recently been forced on Hangouts v17 clients?  I haven't done much digging other than knowing that the pop-up message comes up when you launch the app and is not exitable.  I also know that it uses the following strings, which might help narrowing down where the code to make it pop up is:

```
<string name="forced_upgrade_title">Update Hangouts</string>

<string name="forced_upgrade_body">This version of Hangouts is no longer supported. To continue, please update the app.</string>

<string name="forced_upgrade_update_button">Update now</string>
```

I am trying to do some digging myself but I'm not too well versed in reversing apk's to find something to hook like this with Xposed.

**Comment by AusHick** [comment]
I should mention that this was the last version of Hangouts to support Voice SMS and carrier SMS, which is why I am reluctant to let go of it.

**Comment by ghost** [comment]
XHangouts already restores carrier SMS support, so you could update to Hangouts 20.0.156935076. You'll lose the option to enable merged conversations though.

If you were to compile XHangouts yourself then you could update to Hangouts 24.0.182154523.

**Comment by AusHick** [comment]
My ideal option would be to find out how to hook that pop-up and prevent it from showing up, since merged conversations are kind of important to me.  It doesn't disable any other activity in the app, it's only just a fullscreen block.  Clearing data on Hangouts + Airplane Mode allows you to use the app normally, so it's definitely just some kind of if(version < minapiversion) { popupWindow() }.

My problem is that I just don't really know how to identify that hook in Hangouts since there's no source for it.  My only prior Xposed experience is making an app that prevents the phone from doing critical power shutdowns, which was pretty straight forward since I could just read AOSP.

**Comment by ghost** [comment]
When I get some free time this weekend I'll see if it's possible to restore merged conversations to Hangouts 19 or 18. If I don't have any luck with that then I'll try to patch Hangouts 17.0.148298972 instead.

**Comment by blackbox87** [comment]
https://github.com/blackbox87/XHangouts/releases

**Comment by blackbox87** [comment]
https://github.com/blackbox87/XHangouts/releases

You'll need to uninstall your current version of XHangouts before installing any of mine.

**Comment by kmark** [comment]
For reference, closed because XHangouts v2.17-h18 supports Google Hangouts 18 which should cover this issue.
